src/main/java/de/bayern/gdi/gui/Validator.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * DownloadClient Geodateninfrastruktur Bayern
+ *
+ * (c) 2016 GSt. GDI-BY (gdi.bayern.de)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package de.bayern.gdi.gui;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.reflections.Reflections;
+import org.reflections.scanners.ResourcesScanner;
+import org.reflections.scanners.SubTypesScanner;
+import org.reflections.util.ClasspathHelper;
+import org.reflections.util.ConfigurationBuilder;
+import org.reflections.util.FilterBuilder;
+
+/**
+ * @author Jochen Saalfeld (jochen@intevation.de)
+ */
+
+public class Validator {
+    private Map<String, Reflections> reflectionsList;
+    private static Validator instance;
+
+    private Validator() {
+        this.reflectionsList = new HashMap<String, Reflections>();
+    }
+
+    /**
+     * returns an instance of the class.
+     *
+     * @return the class
+     */
+    public static Validator getInstance() {
```

Comment:
The building of the reflection could take some time, therefore i figured out, it would be good to build them as less as possible